cli/flox/src/commands/install.rs

```diff
@@ -91,6 +91,27 @@ pub struct PkgWithIdOption {
     pub pkg: String,
 }
 
+/// A container for packages during a retry attempt.
+///
+/// If some packages fail to resolve because they lack all of the requested
+/// systems, we retry the installation with only the available systems. In that
+/// case we warn about those packages, but install the other packages as usual.
+/// This container allows us to record whether we should warn or not during
+/// this second installation attempt.
+#[derive(Debug, Clone)]
+struct PackageToInstallRetry {
+    partial: bool,
+    pkg: PackageToInstall,
+}
+
+/// A container for the packages that a user attempted to install.
+#[derive(Debug, Clone)]
+struct PartitionedPackages {
+    successes: Vec<PackageToInstall>,
+    partials: Vec<PackageToInstall>,
+    already_installed: Vec<PackageToInstall>,
+}
```

Comment:
I think we should add another word here to indicate why they are partial. I don't feel strongly about the particular choice of word though.
```suggestion
#[derive(Debug, Clone)]
struct PackageToInstallRetry {
    systems_partial: bool,
    pkg: PackageToInstall,
}

/// A container for the packages that a user attempted to install.
#[derive(Debug, Clone)]
struct PartitionedPackages {
    successes: Vec<PackageToInstall>,
    systems_partials: Vec<PackageToInstall>,
    already_installed: Vec<PackageToInstall>,
}
```

Reply:
Done